CMSIS/NN/Include/arm_nnsupportfunctions.h

```diff
@@ -539,6 +546,18 @@ __STATIC_FORCEINLINE q31_t arm_nn_read_q7x4(const q7_t *in_q7)
     return (val);
 }
 
+/**
+  @brief         Write four q7 to q7 pointer and increment pointer afterwards.
+  @param[in]     in       Double pointer to input value
+  @param[in]     value    Four bytes to copy
+  @return        none
+ */
+__STATIC_FORCEINLINE void arm_nn_write_q7x4_ia(q7_t **in, q31_t value)
```

Comment:
It is very similar to CMSIS-DSP write_q7x4_ia. At some point we should try to identify the common functions we have and export them perhaps to CMSIS-Core.

The only difference is that CMSIS-DSP is still having a #if __ARM_FEATURE_UNALIGNED for the memcpy although memcpy can handle unaligned accesses. But I had observed crashes on M0 with some compilers. I'll need to spend some time on this to understand the root cause and remove this #if But CMSIS-DSP still has to support some older compilers ...

Reply:
Yes, there is scope to combine things here for the bunch of read/write_q7x4/q15x2_ia/no_ia API's . There is a const difference in the input of most when compared to the ones in DSP..